src/css/components/ButtonGroup.scss

```diff
@@ -20,6 +20,9 @@ $ButtonGroup-space: -1 * $border-width-small;
 
 .ButtonGroup > .Button + .Button {
   margin-left: $ButtonGroup-space; // 1
+  &:hover, &.is-active {
+    z-index: 999; //fix issues with overlapping buttons not showing full borders
+  }
```

Comment:
Should this apply to all children buttons rather than just sibling buttons? 

![button-group](https://user-images.githubusercontent.com/9888467/100634780-53d86900-32f5-11eb-865d-0c5636556880.gif)